docs/tutorial/saxpy.rst

```diff
@@ -0,0 +1,780 @@
+    with the Desktop Developemnt Workload. Under Individual Components select:
+
+    - A version of the Windows SDK
+    - "MSVC v143 - VS 2022 C++ x64/x86 build tools (Latest)"
+    - "C++ CMake tools for Windows" (optional)
+
+    .. note::
+
+      The "C++ CMake tools for Windows" individual component is a convenience which
+      puts both ``cmake.exe`` and ``ninja.exe`` onto the PATH inside developer
+      command prompts. You can install these manually, but then you must manage
+      them manually.
+
+    Visual Studio 2017 and later are detectable as COM object instances via WMI.
+    To setup a command line from any shell for the latest Visual Studio's
+    default Visual C++ toolset issue:
+
+    .. code-block:: powershell
+
+      $InstallationPath = Get-CimInstance MSFT_VSInstance | Sort-Object -Property Version -Descending | Select-Object -First 1 -ExpandProperty InstallLocation
+      Import-Module $InstallationPath\Common7\Tools\Microsoft.VisualStudio.DevShell.dll
+      Enter-VsDevShell -InstallPath $InstallationPath -SkipAutomaticLocation -Arch amd64 -HostArch amd64 -DevCmdArguments '-no_logo'
+
+    You should be able to call the compiler on the command line now:
+
+    .. code-block:: powershell
+
+      nvcc --version
+
+Invoking the compiler manually
+------------------------------
+
+To compile and link a single-file application, use the following commands:
+
+.. tab-set::
+  .. tab-item:: Linux and AMD
+    :sync: linux-amd
+
+    .. code-block:: bash
+
+      amdclang++ ./HIP-Basic/saxpy/main.hip -o saxpy -I ./Common -lamdhip64 -L /opt/rocm/lib -O2
+
+  .. tab-item:: Linux and NVIDIA
+    :sync: linux-nvidia
+
+    .. code-block:: bash
+
+      nvcc ./HIP-Basic/saxpy/main.hip -o saxpy -I ./Common -I /opt/rocm/include -O2 -x cu
+
+  .. tab-item:: Windows and AMD
+    :sync: windows-amd
+
+    .. code-block:: powershell
+
+      clang++ .\HIP-Basic\saxpy\main.hip -o saxpy.exe -I .\Common -lamdhip64 -L ${env:HIP_PATH}lib -O2
+
+  .. tab-item:: Windows and NVIDIA
+    :sync: windows-nvidia
+
+    .. code-block:: powershell
+
+      nvcc .\HIP-Basic\saxpy\main.hip -o saxpy.exe -I ${env:HIP_PATH}include -I .\Common -O2 -x cu
+
+Depending on your computer, the resulting binary might or might not run. If not,
+it typically complains about "Invalid device function". That error
+(corresponding to the ``hipErrorInvalidDeviceFunction`` entry of ``hipError_t``)
+means that the runtime could not find a device program binary of the
+appropriate flavor embedded into the executable.
+
+So far, the discussion has covered how data makes it from the host to the
+device and back. It has also discussed the device code as source, with the HIP
+runtime arguing that the correct binary to dispatch for execution. How can you
+find out what device binary flavors are embedded into the executable?
+
+.. tab-set::
+
+  .. tab-item:: Linux and AMD
+    :sync: linux-amd
+
+    The utilities included with ROCm help significantly to inspect binary
```

Comment:
I think binary inspection deserves its own doc, rather then putting it alongside saxpy explanation. In my head person looking for saxpy documentation will be least bothered with code object extraction and the person wanting to play with code objects will be least bothered with saxpy.

Reply:
We can put this feedback to a separate issue. Currently, we do not have binary inspection tutorial and at least we have some here now. 

If we finally have the binary inspection page, we can rethink this suggestion.  Maybe I can put this into a dropdown:
https://sphinx-design.readthedocs.io/en/latest/dropdowns.html

What do you think?

Reply:
I would rather have its own page, playing with code objects might require a bit more detail and populating here will overcomplicate this tutorial.

Reply:
OK. I am creating a separate issue to remove the code inspection from here to a new code inspection tutorial. 

The Boy Scout Rule: Leave the campground cleaner than you found it. → I am merging this in. One tutorial better than zero, and we will improve this later.